crates/factor-key-value/src/lib.rs

```diff
@@ -40,6 +41,7 @@ impl Factor for KeyValueFactor {
     fn init<T: Send + 'static>(&mut self, mut ctx: InitContext<T, Self>) -> anyhow::Result<()> {
         ctx.link_bindings(spin_world::v1::key_value::add_to_linker)?;
         ctx.link_bindings(spin_world::v2::key_value::add_to_linker)?;
+        ctx.link_bindings(spin_world::wasi::keyvalue::store::add_to_linker)?;
```

Comment:
Just a note: We probably don't want to force all embedders of this factor to support wasi key_value. We can leave it as is for now, but we'll probably want to come up with some sort of feature system that allows embedders to choose which interfaces get linked in.

Reply:
As Ryan says, this seems like a system-wide consideration and principle which we should approach from a system-wide perspective.  For example it would apply equally to "embedders shouldn't be forced to support the v1 WITs."  I agree with Michelle, let's capture this but let's not ad-hoc it into each PR.

Reply:
I added the linker bindings for batch and atomic.